authors/apps/authentication/serializers.py

```diff
@@ -23,8 +25,56 @@ class Meta:
         model = User
         # List all of the fields that could possibly be included in a request
         # or response, including fields specified explicitly above.
+        # return a success message on succeesful registration
         fields = ['email', 'username', 'password']
 
+    # https://stackoverflow.com/questions/29813463/django-rest-framework-email-validation
+    def validate_email(self, email_var):
+        # Check if email address is already in use by another user
+        db_email = User.objects.filter(email=email_var)
+        if db_email.exists():
+            raise serializers.ValidationError(
+                "The email address you have used is already registered.")
+        else:
+            return email_var
+
+    # https://stackoverflow.com/questions/29813463/django-rest-framework-email-validation
+    def validate_username(self, username_var):
+        # check if user name already exists in the database
+        # check if username is between 6 to 255 chracters
+        # make sure username can only contain underscore characters included with alphanumeric characters
+        db_username = User.objects.filter(username=username_var)
+        if len(username_var) >= 25:
+            raise serializers.ValidationError(
+                "The username cannot be greater than twenty five characters.")
```

Comment:
Combine the validation for minimum and maximum chars into one

Reply:
Ok @geofrocker, I'll do just that.